src/ast/irbuilderbpf.cpp

```diff
@@ -906,6 +906,19 @@ CallInst *IRBuilderBPF::CreateGetUidGid()
   return createCall(getuidgid_func, {}, "get_uid_gid");
 }
 
+CallInst *IRBuilderBPF::CreateGetNumaId()
+{
+  // u32 bpf_get_numa_node_id(void)
+  // Return: NUMA Node ID
+  FunctionType *getnumaid_func_type = FunctionType::get(getInt64Ty(), false);
```

Comment:
We should probably create a ticket to fix these and all others that claim to be a u32 but are handled as a u64 internally, at least upcast them. What do you think @viktormalik

Reply:
I agree, I ran across a similar case (missing upcast) recently in #2129 and it caused some actual troubles.

Reply:
> I agree, I ran across a similar case (missing upcast) recently in #2129 and it caused some actual troubles.

Do I need to make any changes to the PR/Code?

Reply:
Yes please, change this line to:
```suggestion
  FunctionType *getnumaid_func_type = FunctionType::get(getInt32Ty(), false);
```
and then cast (`zext`) the result returned by the call to 64 bit.

Reply:
Marking as unresolved, the change hasn't been implemented.